### ISSUE TITLE
It Does Not Suport avif format

### ISSUE THREAD

**Issue by Ariel-Schnee** [Feodor2/Mypal68#554]
I keep running into the error that this browser does not support the .avif format.

**Comment by Feodor2** [comment]
Where
Was not going to add it, because it useless same as jpeg-xl

**Comment by Ariel-Schnee** [comment]
> Where Was not going to add it, because it useless same as jpeg-xl

https://valmar.neocities.org/cyoas/gotcyoa/

https://valmar.neocities.org/cyoas/skyrimcyoa/

https://valmar.neocities.org/cyoas/dragonkingdom/

**Comment by Feodor2** [comment]
Okey this is the first site I know use this format, still to add support is not a priority at all.

**Comment by NS-Clone** [comment]
мне вот впервые на днях попалсо такой файл
хз чем его окрыть вообще
не то что в браузере...

**Comment by zanud** [comment]
> ...или это раз оно на сайтах
> это значит что внахром уже прикрутили?

В Википедии пишут, что Chrome поддерживает с 2020 года, а Firefox с 2021-го (версия 93, хотя в скрытом виде появилось ещё в v.89).

**Comment by NS-Clone** [comment]
с этим авифом таже фехня что и вебпой
открываешь скачанный авиф
а внтури жипег...

...ну и наверно через пару лет
этот авиф таки придетсо впиливать в браузер

за предыдущие 2 года у меня скачано 7 авифов
за 20 дней 25-го года умя скачано 10 авифов

**Comment by NS-Clone** [comment]
@Feodor2 а через какую либу фаирфокс кушает avif-ы?

libavif можно собрать со свежим
libaom-3.12.0
который в свою очередь можно собрать с set(AOM_TARGET_CPU "generic")
и для которого нужон только perl (5.28.0 работать)
и если потом подменить вызовы kernel32 на dll-ку от спepмиума
оно дажо декодирует какието av1 видева
как проверить сам libavif яхз
да и пока получилось собрать только shared

**Comment by NS-Clone** [comment]
а собственно нате тошо есть

[libavif-1.1.1__cpu_generic__gcc_12.2.0_dwarf_XP_Patch_r0000.zip](https://github.com/user-attachments/files/18951782/libavif-1.1.1__cpu_generic__gcc_12.2.0_dwarf_XP_Patch_r0000.zip)
работоспособность неизвестна

**Comment by zanud** [comment]
Два вопроса:

1. Насколько этот файл может быть полезен для программы, собираемой совершенно другим компилятором?

2. Разве gcc12.2.0 для XP существует? Если да, то где его взять?

**Comment by NS-Clone** [comment]
>     2. Разве gcc12.2.0 для XP существует? Если да, то где его взять?

тут от niXman-а

**Comment by zanud** [comment]
> dll-ку можно присобачить куда угодно

Нет, конечно. Есть различия соглашения о вызовах. Есть различия в ABI. И даже если всё совпадает, DLL может нуждаться в куче других DLL от компилятора.

Я уже привык к тому, что если в названии есть название компилятора и его версия, то оно именно для этого компилятора и предназначено - и ни для какого другого.

**Comment by NS-Clone** [comment]
> Я уже привык к тому, что если в названии указан компилятор и его версия, то оно именно для этого компилятора и предназначено - и ни для какого другого.

ну это на особо клинических случаях
вон wxvidgets 2.x даже проверяет чета при запуске
и отказываетсо собиратсо
в итоге его надо блджд собирать своим компилером
а код gcc5 никрена не компилетсо на gcc10
какие настройки совместимости на ставь...

**Comment by zanud** [comment]
> это там где "гарантированно" буит работать

Да. И одновременно это означвает, что с чем угодно другим может не заработать.
У Microsoft на сайте здоровенная статья есть, где рассказывается о взаимных совместимостях-несовместимостях разных версих их VC.

> вот вроде нашел

Я оттуда только что уже 12.2 скачал. Попробовал - запускается. Но пытаться компилировать буду завтра-послезавтра.

**Comment by NS-Clone** [comment]
> Да. И одновременно это означает, что с чем угодно другим может не заработать.

по сравнению с всегда нерабочим красноглазым питоном
которому нужно всё и вся с точностью +- 0.0.0.1 версий
у нас тут все работает просто отлишно 👍

**Comment by zanud** [comment]
> ну это на особо клинических случаях

Нет, это норма жизни.

Причём это мы ещё говорим только о библиотеках, собранных из обычного Сишного кода.
Как только возникают объекты, несовместимыми могут стать даже библиотеки одного и того же компилятора, если собраны с разными параметрами.

**Comment by NS-Clone** [comment]
> Причём это мы ещё говорим только о библиотеках, собранных из обычного Сишного кода. Как только возникают объекты, несовместимыми могут стать даже библиотеки одного и того же компилятора, если собраны с разными параметрами.

ну кто то писал что якобы нельзя комбинировать
-OFast (и вообще его юзать) с другими -О
я уся его комбинирую и у меня все работаит (С)

**Comment by zanud** [comment]
> ну кто то вообще писал что якобы нельзя комбинировать -OFast (и вообще его юзать) с другими -О я уся его комбинирую и у меня все работаит (С)

-O это оптимизация. Насколько я понимаю, на взаимодействие разных модулей между собой она не влияет.

**Comment by NS-Clone** [comment]
> -O это оптимизация. Насколько я понимаю, на взаимодействие разных модулей между собой она не влияет.

она форсит включение алигнов для всего на больших значениях O
и якобы такие могут не ожидать их отсутствия

**Comment by NS-Clone** [comment]
> Что, и Z80 имеешь? Может даже с кнопкой?

может даже с 40+

**Comment by zanud** [comment]
> она форсит включение алигнов для всего на больших значениях O

Это и есть часть ABI. Но выравнивание на что? По умолчанию выравнивание делается на размер машинного слова. Как это можно испортить?

> может даже с 40+ кнопков

Не понял. Ты что, не знаешь, что такое "кнопка" у i8080/Z80?

**Comment by NS-Clone** [comment]
> Потому и спрашиваю, что удивился, когда увидел, что GCC-шную библиотеку предлагают для VC.
а я знаю чем собирает feodor2?

да и вполне встречаютсо софты
в которых насапысны и vscrt и gcc-шные либы

**Comment by NS-Clone** [comment]
> Это и есть часть ABI. Но выравнивание на что? По умолчанию выравнивание делается на размер машинного слова. Как это можно испортить?

какое еще машинное слово?
ты на арме?
ну вот если ты захочешь юзать что либо через SSE
там тебе настрочит каких нить align 64 или около того...

**Comment by zanud** [comment]
> а я знаю чем собирает feodor2?

В Mypal только VC-шные DLL, и ни одной GCC-шной.

Нет, я так умею сделать. Но это будет крошечная, мучительно написанная программа.

> какое еще машинное слово?

В нашем случае - 32-разрядное.

> ты на арме?

Я на чём угодно. Понятие "машинное слово" универсальное, от процессора не зависит. если уж на то пошло, оно старше понятия "байт", а Intel своим WORD вообще всю терминологию перекорёжила и испоганила.

**Comment by NS-Clone** [comment]
> В нашем случае - 32-разрядное.

и какой размер машинного слова на avx512?

**Comment by NS-Clone** [comment]
> В Mypal только VC-шные DLL, и ни одной GCC-шной.

ну а так feodor2 может собрать и себе сам
и дажо лучшо

**Comment by NS-Clone** [comment]
хотя запустил какойто последний vlc
(почти из коробки)
а он вполне успешно улетает на av1

**Comment by NS-Clone** [comment]
@Feodor2 
а вощем визуально похоже что в самом libavif от вин15 ничего нет
в libaom 2 функции
InitOnceBeginInitialize
InitOnceComplete
в aom_once.h

и там же есть HAVE_PTHREAD_H альтернатива и not thread safe версия

ну и парочка _WIN32_WINNT=0x0601
и видимо асмовставки которые вполне отключаютсо одной командой

хз чем протестить готовую либу

**Comment by NS-Clone** [comment]
а вот ищо что попалось
https://github.com/Kagami/avif.js
хз что это конешно

**Comment by NS-Clone** [comment]
там где avif там и av1
mypal тоже его есное дело что не крутит

вот закину в качестве примера тестовое видео
мож кому пригодится
[spbtv_sample_bipbop_av1_960x540_25fps.zip](https://github.com/user-attachments/files/20276614/spbtv_sample_bipbop_av1_960x540_25fps.zip)

supermium оно крутит

**Comment by NS-Clone** [comment]
а нельзя натравить нейронку 
чтобы лучшо примержить pull request с avif-ом?
https://github.com/Feodor2/Mypal68/pull/596
